__tests__/app.test.js

```diff
@@ -116,14 +116,78 @@ describe('GET /api/reviews/:review_id', () => {
             });
         });
         });
-        test('404: non-existent route', ()=>{
+    });
+
+    describe('POST /api/reviews/:review_id/comments',() => {
+        test('201: responds with comments newly added to the database', () => {
+            const comment1 = {
+                username: 'mallionaire',
+                body: 'Loving it!'
+            };
+            const reviewId = 2
             return request(app)
-            .get('/apz')
-            .expect(404)
-            .then(({body})=>{
+            .post(`/api/reviews/${reviewId}/comments`)
+            .send(comment1)
+            .expect(201)
+            .then((result) => {
+            expect(result.body).toMatchObject({
+                    author: 'mallionaire',
+                    body: 'Loving it!',
+                    review_id: 2,
+                    comment_id: 7,
+                    votes:0,
+                    created_at: "2017-11-22T12:43:33.389Z",
+            })
+            console.log(result.body, 141)
+            });
+            });
+            });
+        test('400: when a key is null', () => {
+            const reviewId = 'z'
+            return request(app)
+            .get(`/api/reviews/${reviewId}`)
+            .expect(400)
+            .then(({body}) => {
+            expect(body).toEqual({msg: 'path not found!'});
+            });
+            });
+        test('201: extra keys in the request object should be ignored', () => {
+            const reviewId = 'z'
+            return request(app)
+            .get(`/api/reviews/${reviewId}`)
+            .expect(400)
+            .then(({body}) => {
             expect(body).toEqual({msg: 'path not found!'});
             });
+            });
+        test ('extra keys in the request object should be ignored', () => {
+            const comment2 = {
+                username: 'mallionaire',
+                body: 'Psych!',
+                role: 'admin'
+            };
+            const reviewId = 2
+            return request(app)
+            .post(`/api/reviews/${reviewId}/comments`)
+            .send(comment2)
+            .expect(201)
+            .then((result) => {
+            expect(result.body).toMatchObject({
+                    author: 'mallionaire',
+                    body: 'Psych!',
+                    review_id: 2,
+                    comment_id: 7,
+                    votes:0,
+                    created_at: "2017-11-22T12:43:33.389Z",
+            });
         });
-
-    });
-
+        });
+        test('400: invalid review_id', () => {
+            const reviewId = 'z'
+            return request(app)
+            .get(`/api/reviews/${reviewId}`)
+            .expect(400)
+            .then(({body}) => {
+            expect(body).toEqual({msg: 'path not found!'});
+            });
+            });
```

Comment:
Could we rework these tests please we seem to have identical tests with different descriptions and are missing the following situations:
- non-existent ID
- Missing required fields or empty input body
- username doesn't exist